tmva/tmva/test/DNN/Utility.h

```diff
@@ -208,6 +208,28 @@ inline T relativeError(const T &x, const T &y)
   return diff / (abs(x) + abs(y));
 }
 
+//______________________________________________________________________________
+template <>
+inline Double_t relativeError(const Double_t &x, const Double_t &y)
+{
+   if ((std::abs(x) > 1e-8) && (std::abs(y) > 1e-8)) {
+      return std::fabs((x - y) / y);
+   } else {
+      return std::fabs(x - y);
+   }
+}
+
+//______________________________________________________________________________
+template <>
+inline Float_t relativeError(const Float_t &x, const Float_t &y)
+{
+   if ((std::abs(x) > 1e-8) && (std::abs(y) > 1e-8)) {
+      return std::fabs((x - y) / y);
+   } else {
+      return std::fabs(x - y);
+   }
+}
```

Comment:
Hi @simonpf I don't think this is the right solution. This duplicates the code that already exists in the generic template, and uses a single hard-coded value of epsilon for both single and double precision, which is almost guaranteed to be inadequate for one of them. Please note also that the existing code does protect against divisions by small numbers, although it may be a bit too strict in some cases. The correct solution is to add epsilon as a parameter to the generic template with a default value, and change the default if it is too strict for a particular test.